src/components/VideoPlayer.vue

```diff
@@ -0,0 +1,95 @@
+<template>
+  <div class="video-player">
+    <template v-if="!playingUrl">
+      <img :src="thumbnailUrl" @click="onClickVideo(videoUrl)" :style="styleObject" />
+      <span class='icon-play' @click="onClickVideo(videoUrl)"></span>
+    </template>
+    <video
+      v-show="playingUrl"
+      ref="video"
+      :src="playingUrl"
+      controls=true
+      autoplay=true
+      type="video/mp4" />
+  </div>
+</template>
+
+<script>
+const purposeTypes = {
+  curriculumOverview: { width: '100%' },
+  curriculumStep: { width: '100%', 'border-radius': '8px' }
+}
+
+export default {
+  name: 'video-player',
+  props: {
+    purpose: {
+      default: 'curriculumOverview',
+      validator: function(value) {
+        return Object.keys(purposeTypes).indexOf(value) !== -1
+      }
+    },
+    videoUrl: { type: String },
```

Comment:
`videoUrl` and `thumbnailUrl` should have their default value

Reply:
Otherwise it must be required

Reply:
I decided to add required.